curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/645f7879ebbdb201892e55e1.md

```diff
@@ -26,7 +26,12 @@ assert(code.match(/const\s+userAvatarUrl\s*=/));
 You should assign a `ternary` operator to the `userAvatarUrl`. Your `ternary` should check if `avatar` starts with `/user_avatar/` and return `avatarUrl.concat(avatar)` or `avatar`.
 
 ```js
-assert(code.match(/const\s+userAvatarUrl\s*=\s*avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/));
+const hasStartsWith = code.includes('startsWith');
+const regex = hasStartsWith
+   ? /const\s+userAvatarUrl\s*=\s*avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/
+   : /const\s+userAvatarUrl\s*=\s*avatar\.test\(\s*\/(?:\^)?user_avatar\/\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/;
```

Comment:
There's slight issue with this - string doesn't have `test` method, so code that would match this regex causes error: `[TypeError: avatar.test is not a function]`

Reply:
Yeah, that's right, thanks for the suggestion, I'll make the change accordingly.